kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -280,6 +280,31 @@ ExpressionTransformResult visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ExpressionTransformResult visitLike(final Predicate like) {
+            List<ExpressionTransformResult> children =
+                    like.getChildren().stream()
+                        .map(this::visit)
+                        .collect(Collectors.toList());
+            int size = children.size();
+            if (size<2 || size>3) {
```

Comment:
```suggestion
            if (size < 2 || size > 3) {
```

Reply:
Please add spaces around operators here and elsewhere in the PR.

Reply:
You can actually move this check to `LikeExpressionEvaluator.validateAndTransform()`. With that all checks are in one place.